rero_ils/modules/patron_transactions/api.py

```diff
@@ -103,156 +106,57 @@ def update(self, data, commit=True, dbcommit=True, reindex=True):
-        results = query.source('pid').scan()
-        for result in results:
-            yield result.pid
-
-    @classmethod
-    def get_transactions_total_amount_for_patron(cls, patron_pid, status=None,
-                                                 types=None,
-                                                 with_subscription=True):
-        """Get total amount transactions linked to a patron.
-
-        :param patron_pid: the patron pid being searched
-        :param status: (optional) transaction status filter,
-        :param types: (optional) transaction type filter,
-        :param with_subscription: (optional) include or exclude subscription
-        type filter.
-        :return: return total amount of transactions.
-        """
-        search = cls._build_transaction_query(patron_pid, status, types)
-        if not with_subscription:
-            search = search.exclude('terms', type=['subscription'])
-        search.aggs.metric('transactions_total_amount',
-                           'sum', field='total_amount')
-        # set the from/size to 0
-        search = search[0:0]
-        results = search.execute()
-        return results.aggregations.transactions_total_amount.value
-
-    @classmethod
-    def get_transactions_count_for_patron(cls, patron_pid, status=None):
-        """Get patron transactions count linked to a patron.
-
-        :param patron_pid: the patron pid being searched
-        :param status: (optional) transaction status filter,
-        """
-        query = cls._build_transaction_query(patron_pid, status)
-        return query.source().count()
-
-    @classmethod
-    def get_transactions_pids_for_patron(cls, patron_pid, status=None):
-        """Get patron transactions pids linked to a patron.
-
-        :param patron_pid: the patron pid being searched
-        :param status: (optional) transaction status filter,
-        """
-        query = cls._build_transaction_query(patron_pid, status)
-        return sorted_pids(query)
-
-    @classmethod
-    def get_transactions_by_patron_pid(cls, patron_pid, status=None):
-        """Return opened fees for patron.
+    # GETTER & SETTER =========================================================
+    #   * Define some properties as shortcut to quickly access object attrs.
+    #   * Defines some getter methods to access specific object values.
 
-        :param patron_pid: the patron PID
-        :param status: the status of transaction
-        :return: return an array of transaction
-        """
-        query = cls._build_transaction_query(patron_pid, status)
-        return query\
-            .params(preserve_order=True)\
-            .sort({'creation_date': {'order': 'asc'}})\
-            .scan()
-
-    @classmethod
-    def get_last_transaction_by_loan_pid(cls, loan_pid, status=None):
-        """Return last fee for loan.
+    @property
+    def loan_pid(self):
+        """Get the `Loan` pid related to this transaction."""
+        if self.get('loan'):
```

Comment:
Maybe in one line?

Reply:
Some transaction type isn't related to a loan (photocopy, subscription), so the "loan" dict key isn't available. Calling `extracted_data_frol_ref()` with None will raise an error.